packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -14,6 +14,15 @@ function aliasIdToElementId(args, props) {
   }
 }
 
+// Use `_targetObject` to avoid stomping on a CP
+// that exists in the component
+function privatizeTargetObject(props) {
+  if (props.targetObject) {
+    props._targetObject = props.targetObject;
+    delete props.targetObject;
```

Comment:
Not sure if want to delete off of props as I think it's going to cause `props` to go into dictionary mode and make it's lookups much slower. However if it's already a `dict` it should be fine. Can you look `processedArgs.value()` actually evaluates to in glimmer?